Dockerfile

```diff
@@ -0,0 +1,109 @@
+# ===============================================
+FROM helsinkitest/node:12-slim as appbase
+# ===============================================
+
+# Offical image has npm log verbosity as info. More info - https://github.com/nodejs/docker-node#verbosity
+ENV NPM_CONFIG_LOGLEVEL warn
+
+# Global npm deps in a non-root user directory
+ENV NPM_CONFIG_PREFIX=/app/.npm-global
+ENV PATH=$PATH:/app/.npm-global/bin
+
+# Set env
+ARG PORT
+ENV PORT $PORT
+
+# Use non-root user
+USER appuser
+
+# Copy package.json and package-lock.json/yarn.lock files
+COPY --chown=appuser:appuser package*.json *yarn* ./
+
+# Copy internals becaused they are called in preinstall and postinstall
+# hooks
+COPY --chown=appuser:appuser internals/ ./internals/
+
+# Install npm depepndencies
+ENV PATH /app/node_modules/.bin:$PATH
+
+# =============================
+FROM appbase as development_base
```

Comment:
As `development_base` is extended and nothing is really copied from the stage, having both `development_base` and `development` is redundant in this case and only the `development` stage would suffice.